libs/core/algorithms/include/hpx/parallel/unseq/simd_helpers.hpp

```diff
@@ -0,0 +1,161 @@
+//  Copyright (c) 2022 A Kishore Kumar
+//  Copyright (c) 2023 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/functional/detail/invoke.hpp>
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/parallel/unseq/reduce_helpers.hpp>
+#include <hpx/type_support/construct_at.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+// Please use static assert and enforce Iter to be Random Access Iterator
+namespace hpx::parallel::util {
+    /*
+        Compiler and Hardware should also support vector operations for IterDiff,
+        else we see slower performance when compared to sequential version
+    */
```

Comment:
What does this comment mean? Should that requirement be somehow enforced?

Reply:
As discussed [previously](https://github.com/STEllAR-GROUP/hpx/pull/6286#discussion_r1239159849), changing data types can lead to vectorization being broken. I wanted to leave a note for the future regarding it. Use perf regression tests might not be the best way to do this as they often seem to have lot of variance in them.

Reply:
Well, a comment doesn't help with this, does it? I think setting up performance tests would be the better option. @Pansysk75 can help with that.

Reply:
@hkaiser Do you mean automated ones that report on GH? He asked me about this but I discouraged him from doing that yet. It can certainly be done though, if that's what you have in mind

Reply:
Well, I thought we could add it to our performance tests.

Reply:
added it a separate [PR](https://github.com/STEllAR-GROUP/hpx/pull/6325)